Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
I do think you're right though. This works based on the coincidence that a request life cycle matches a thread life cycle. If the concurrency model were to change such that this is not true this system would fail. 

On the other hand I don't know if that's something that would ever happen, or if it would even be difficult to fix. Concurrency model changing would likely break a lot of things.